src/server/pfs/server/commit_store.go

```diff
@@ -0,0 +1,145 @@
+package server
+
+import (
+	"context"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/track"
+)
+
+type commitStore interface {
+	AddFileset(ctx context.Context, commit *pfs.Commit, filesetID fileset.ID) error
+	GetFileset(ctx context.Context, commit *pfs.Commit) (filesetID *fileset.ID, err error)
+	SetFileset(ctx context.Context, commit *pfs.Commit, id fileset.ID) error
+	DropFilesets(ctx context.Context, commit *pfs.Commit) error
+}
+
+var _ commitStore = &postgresCommitStore{}
+
+type postgresCommitStore struct {
+	db *sqlx.DB
+	s  *fileset.Storage
+	tr track.Tracker
+}
+
+func newPostgresCommitStore(db *sqlx.DB, tr track.Tracker, s *fileset.Storage) *postgresCommitStore {
+	return &postgresCommitStore{
+		db: db,
+		s:  s,
+		tr: tr,
+	}
+}
+
+func (cs *postgresCommitStore) AddFileset(ctx context.Context, commit *pfs.Commit, id fileset.ID) error {
+	// clone to remove the ttl.
+	id2, err := cs.s.Clone(ctx, id, track.NoTTL)
+	if err != nil {
+		return err
+	}
+	var num int
+	if err := cs.db.GetContext(ctx, &num,
+		`INSERT INTO pfs.commit_diffs (repo_name, commit_id, fileset_id)
+		VALUES ($1, $2, $3)
+		RETURNING num
+	`, commit.Repo.Name, commit.ID, *id2); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cs *postgresCommitStore) GetFileset(ctx context.Context, commit *pfs.Commit) (*fileset.ID, error) {
+	id, err := cs.getTotal(ctx, commit)
+	if err == nil {
+		return cs.s.Clone(ctx, *id, defaultTTL)
+	}
+	ids, err := cs.getDiff(ctx, commit)
```

Comment:
The diff + parent total would be the fileset for the commit. This seems to just return the diff when the total does not exist, so I think we need to grab the parent total and add it to the composition.

Reply:
So I fixed this, and now we can read open commits.  But it is potentially expense. We can also read commits with open parents, and grandparents, etc. which is potentially even more expensive.